packages/database/src/core/AppCheckTokenProvider.ts

```diff
@@ -30,17 +31,26 @@ import { warn } from './util/util';
  */
 export class AppCheckTokenProvider {
   private appCheck?: FirebaseAppCheckInternal;
+  private serverAppAppCheckToken?: string;
+  private appName: string;
   constructor(
-    private appName_: string,
+    app: FirebaseApp,
     private appCheckProvider?: Provider<AppCheckInternalComponentName>
   ) {
+    this.appName = app.name;
+    if (_isFirebaseServerApp(app) && app.settings.appCheckToken) {
+      this.serverAppAppCheckToken = app.settings.appCheckToken;
+    }
```

Comment:
During initialization of the `AppCheckTokenProvider` that RTDB implements, check to see if the provided instance of `FirebaseApp` is one of `FirebaseServerApp`, and if it contains an App Check token, then store it locally so that it can be returned by this provide when its `getToken` method is invoked (below).